### PR TITLE
Add coverage RDD

### PR DESCRIPTION
**Pull request by Georgehe4**
Added comments, Unit tests pass

### CODE REVIEW

File: mango-cli/src/main/webapp/resources/pileup.js [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit faaf7b789966e5d289cd0e12bdec9bef10266a99
+Subproject commit 54747798e4108ff6d1b6b40905356347d7a5f077
```
Comment (akmorrow13) [inline mango-cli/src/main/webapp/resources/pileup.js]: remove this from your commit if you didnt change pileup

File: mango-core/src/main/scala/org/bdgenomics/mango/models/CoverageMaterialization.scala [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+   * Formats raw data from KLayeredTile to JSON. This is required by KTiles
+   *
+   * @param data RDD of (id, AlignmentRecord) tuples
+   * @return JSONified data
+   */
+  def stringify(data: RDD[(String, Coverage)]): Map[String, String] = {
+    val flattened: Map[String, Array[PositionCount]] = data
+      .collect
+      .groupBy(_._1)
+      .map(r => (r._1, r._2.map(_._2)))
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/models/CoverageMaterialization.scala]: I would do these map stages before the collect, then do the groupBy

File: mango-core/src/main/scala/org/bdgenomics/mango/models/LazyMaterialization.scala [hunk 3]
```diff
@@ -141,7 +141,8 @@ abstract class LazyMaterialization[T: ClassTag] extends Serializable with Loggin
         val d = load(region, fp).map(v => (k, v))
         data = data.union(d)
       })
-
+      data.cache()
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/models/LazyMaterialization.scala]: Is there a reason for this?

File: mango-core/src/test/scala/org/bdgenomics/mango/models/CoverageMaterializationSuite.scala [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  sparkTest("create new CoverageRecordMaterialization") {
+    val lazyMat = CoverageRecordMaterialization(sc, files, dict)
+  }
+
+  sparkTest("return coverage from CoverageRecordMaterialization") {
+    val data = CoverageRecordMaterialization(sc, files, dict)
+    val region = new ReferenceRegion("chrM", 0L, 20L)
+    val freq = data.getCoverage(region).get(key).get
+    val coverage = parse(freq).extract[Array[PositionCount]]
+    // extract number of positions in string ('position' => 'p')
```
Comment (akmorrow13) [inline mango-core/src/test/scala/org/bdgenomics/mango/models/CoverageMaterializationSuite.scala]: remove this comment it looks like a copy paste

File: mango-core/src/test/scala/org/bdgenomics/mango/models/CoverageMaterializationSuite.scala [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    assert(coverage.length == region.length())
+  }
+
+  sparkTest("return coverage overlapping multiple materialized nodes") {
+    val data = CoverageRecordMaterialization(sc, files, dict)
+    val region = new ReferenceRegion("chrM", 90L, 110L)
+    val freq = data.getCoverage(region).get(key).get
+    val coverage = parse(freq).extract[Array[PositionCount]].sortBy(_.position)
+    // extract number of positions in string ('position' => 'p')
```
Comment (akmorrow13) [inline mango-core/src/test/scala/org/bdgenomics/mango/models/CoverageMaterializationSuite.scala]: remove comment

File: mango-cli/src/main/webapp/resources/pileup.js (hunk 1 above)
Comment (Georgehe4) [inline mango-cli/src/main/webapp/resources/pileup.js]: Pileup version has updated now.

Comment (akmorrow13) [inline mango-cli/src/main/webapp/resources/pileup.js]: ok great

Comment (akmorrow13) [inline mango-cli/src/main/webapp/resources/pileup.js]: ok perfect. leave this in then

File: mango-core/src/main/scala/org/bdgenomics/mango/models/LazyMaterialization.scala [hunk 6]
```diff
@@ -141,7 +141,6 @@ abstract class LazyMaterialization[T: ClassTag] extends Serializable with Loggin
         val d = load(region, fp).map(v => (k, v))
         data = data.union(d)
       })
-
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/models/LazyMaterialization.scala]: remove this if not necessary

**Pull request closed without merging**